### PR TITLE
API-2532: Fixed Response model error.

### PR DESCRIPTION
**Pull request by googley42**

### CODE REVIEW

File: app/uk/gov/hmrc/apisubscriptionfields/controller/FieldsDefinitionController.scala [hunk 1]
```diff
@@ -49,8 +49,8 @@ class FieldsDefinitionController @Inject() (service: FieldsDefinitionService) ex
 
   def getAllFieldsDefinitions: Action[AnyContent] = Action.async { implicit request =>
     Logger.debug(s"[getAllFieldsDefinitions]")
-    service.getAll.map(list =>
-      Ok(Json.toJson(list))
+    service.getAll.map(fieldsDefinitionResponse =>
```
Comment (gokyo) [inline app/uk/gov/hmrc/apisubscriptionfields/controller/FieldsDefinitionController.scala]: write in a single line:
```
service.getAll.map(Ok(Json.toJson(_)))
```

File: app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala [hunk 2]
```diff
@@ -41,11 +41,14 @@ class FieldsDefinitionService @Inject() (repository: FieldsDefinitionRepository)
...
   }
 
-  def getAll: Future[List[FieldsDefinitionResponse]] = {
+  def getAll: Future[FieldsDefinitionResponse] = {
     Logger.debug(s"[getAll]")
-    for {
+    (for {
       list <- repository.fetchAll()
-    } yield list.map(asResponse)
+    } yield list map
```
Comment (gokyo) [inline app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala]: Too many lines of code
I am sure you can write it in a more concise way

File: app/uk/gov/hmrc/apisubscriptionfields/controller/FieldsDefinitionController.scala (hunk 1 above)
Comment (googley42) [reply app/uk/gov/hmrc/apisubscriptionfields/controller/FieldsDefinitionController.scala]: suggested code does not work - probably needs a type annotation to make it work - so I would suggest original style is OK

File: app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala (hunk 2 above)
Comment (googley42) [reply app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala]: I have simplified to 
``` scala
    (for {
      list <- repository.fetchAll()
    } yield list map (_.fieldDefinitions)) map (_.flatten) map (FieldsDefinitionResponse(_))
```

File: app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala [hunk 3]
```diff
@@ -41,11 +41,11 @@ class FieldsDefinitionService @Inject() (repository: FieldsDefinitionRepository)
...
   }
 
-  def getAll: Future[List[FieldsDefinitionResponse]] = {
+  def getAll: Future[FieldsDefinitionResponse] = {
     Logger.debug(s"[getAll]")
-    for {
+    (for {
       list <- repository.fetchAll()
-    } yield list.map(asResponse)
+    } yield list map (_.fieldDefinitions)) map (_.flatten) map (FieldsDefinitionResponse(_))
```
Comment (gokyo) [inline app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala]: Why do not you try to optimise this code?

**Comment by googley42** [comment]
:+1:

**Pull request merged**